agent/configentry/resolve.go

```diff
@@ -161,9 +162,12 @@ func ComputeResolvedServiceConfig(
 			// upstreams that are inferred from intentions and do not have explicit upstream configuration.
 			cfgMap := make(map[string]interface{})
 			upstreamDefaults.MergeInto(cfgMap)
+			if !args.MeshGateway.IsZero() {
+				cfgMap["mesh_gateway"] = args.MeshGateway
```

Comment:
This is weird. If we leave the old logic in-place, then `NodeService.Proxy.MeshGateway` will NOT override the value from `ServiceDefaults.UpstreamConfig.Defaults` for the wildcard upstream.

This means that a user of tproxy could encounter a situation where they configure `NodeService.Proxy.MeshGateway` with a value that is ignored for all implicit upstreams, but honored for explicit ones.